tutorial-review.md

```diff
@@ -0,0 +1,1053 @@
+import Nav from 'react-bootstrap/Nav';
+import Container from 'react-bootstrap/Container';
+
+function App() {
+  return (
+    <>
+      <Navbar bg="dark" expand="lg" variant="dark">
+        <Container>
+          <Navbar.Brand href="/">My Blog</Navbar.Brand>
+          <Navbar.Toggle aria-controls="basic-navbar-nav" />
+          <Nav className="me-auto">
+            <Nav.Link href="/posts/new">New</Nav.Link>
+          </Nav>
+        </Container>
+      </Navbar>
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/posts/:id" element={<Post />} />
+        <Route path="/posts/new" element={<Create />} />
+        <Route path="/posts/:id/edit" element={<Edit />} />
+      </Routes>
+    </>
+  );
+}
+
+export default App;
+```
+
+### Axios client
+
+Our client will have to make API calls to the server to perform operations on the database, this is why we installed `axios` at the beginning of this tutorial. It is now time to use it, however we can wrap inside a `http` library file and export it as a module:
+
+1. We need to take into account that making API calls in local is like calling a different server as client and servers run on different ports, a complete different configuration compared to the deployment on Koyeb later on.
+2. The HTTP object is exported along with the basic methods to call GET, POST, PUT and DELETE endpoints.
+
+Create a new folder `/lib` and inside add `http.js`
+
+```bash
+  mkdir lib
+  touch /lib/http.js
+```
+
+Add the following code:
+
+```javascript
+import axios from 'axios';
+// When building the client into a static file, we do not need to include the server path as it is returned by it
+const domain = process.env.NODE_ENV === 'production' ? '' : 'http://localhost:3001';
+
+const http = (
+  url,
+  {
+    method = 'GET',
+    data = undefined,
+  },
+) => {
+  return axios({
+    url: `${domain}${url}`,
+    method,
+    data,
+  });
+};
+
+// Main functions to handle different types of endpoints
+const get = (url, opts = {}) => http(url, { ...opts });
+const post = (url, opts = {}) => http(url, { method: 'POST', ...opts });
+const put = (url, opts = {}) => http(url, { method: 'PUT', ...opts });
+const deleteData = (url, opts = {}) => http(url, { method: 'DELETE', ...opts });
+
+const methods = {
+  get,
+  post,
+  put,
+  delete: deleteData,
+};
+
+export default methods;
+```
+
+We will see how to use the `http` object in the next section.
+
+### Create containers and reusable components
+
+The very first component we are going to build is `Home`, in charge of rendering the list of posts as well as the header of the homepage.
+To render the list of posts Home has to 
+1. Call the server GET `/api/posts` endpoint after the first rendering
+2. Store the array posts in the state
+3. Render the posts to the user and link them to `/posts/:post_id` to read the content
+
+In the terminal create a folder `/pages` and a file `home.js` in it:
```

Comment:
Does this go inside the `mongo-blog` project or elsewhere? If elsewhere, can you add an explanation to guide the reader with what is happening?

Reply:
I clarified it should be created under `/client`, let me know